S2VX.Game/Story/Note/GameHoldNote.cs

```diff
@@ -30,66 +21,66 @@ public class GameHoldNote : HoldNote {
         [Resolved]
         private PlayScreen PlayScreen { get; set; }
 
-        private const int HitMissThreshold = 200;
-        private const int ReleaseMissThreshold = 200;
-        private int HitTimingError;
-        private int ReleaseTimingError;
-
+        private const int MissThreshold = 200;
+        private SampleChannel Hit { get; set; }
+        private SampleChannel Miss { get; set; }
+        private double TimeOfLastUpdate { get; set; }
+        private int ScoreBefore { get; set; }
+        private int ScoreDuring { get; set; }
+        private int ScoreAfter { get; set; }
+        private HoldNoteState HoldNoteState { get; set; } = HoldNoteState.NotVisibleBefore;
         private Key? KeyBeingHeld { get; set; }
         private MouseButton? MouseButtonBeingHeld { get; set; }
         private bool ShouldBeRemoved { get; set; }
 
+        [BackgroundDependencyLoader]
+        private void Load(AudioManager audio) {
+            Hit = audio.Samples.Get("hit");
+            Miss = audio.Samples.Get("miss");
+        }
 
-        private void Delete() => ShouldBeRemoved = true;
-
-        private void RecordMiss() {
-            var missThreshold = HitMissThreshold;
-            ScoreInfo.AddScore(missThreshold);
-            PlayScreen.PlayInfoBar.RecordHitError(missThreshold);
-            Miss.Play();
-            Delete();
+        private void Delete() {
+            ScoreBefore = Math.Clamp(ScoreBefore, 0, MissThreshold);
+            ScoreAfter = Math.Clamp(ScoreAfter, 0, MissThreshold);
+            var totalScore = ScoreBefore + ScoreDuring + ScoreAfter;
+            ScoreInfo.AddScore(totalScore);
```

Comment:
It makes more sense to me to calculate/show the score in real time, or are there other implications for that?

Reply:
Sure I can have the score update in real time but the HitError bar at the top right now is not designed for updating so that will have to stay the way it is

Reply:
I see, we can add together for now and add a task for updating in real time

Reply:
Created #288 for adding real-time functionality for the HitError bar